heaps/min_heap.py

```diff
@@ -57,18 +68,40 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: (log n)
+            Space complexity: (O 1)
         """
-        pass
+        if index == 0:
+                return index
+
+        parent_node = (index - 1) // 2
+        current_size = self.store
+        if current_size[parent_node].key > current_size[index].key:
+            self.swap(parent_node, index)
+            self.heap_up(parent_node)
 
     def heap_down(self, index):
         """ This helper method takes an index and 
             moves the corresponding element down the heap if it's 
             larger than either of its children and continues until
             the heap property is reestablished.
         """
-        pass
+        current_size = self.store
+        left_c = index * 2 + 1
+        right_c= index * 2 + 2
+        
+        if left_c < len(self.store):
```

Comment:
👀 You made your alias to `self.store`, so be sure to use it here for consistency. The biggest concern around consistency is that when we do things inconsistently, it makes the person reading the code stop and wonder whether the inconsistency is meaningful, or just an oversight.

```py
        if left_c < len(current_size):
```